src/engine/cuecontrol.cpp

```diff
@@ -340,13 +323,16 @@ void CueControl::hotcueSet(HotcueControl* pControl, double v) {
         return;
 
     int hotcue = pControl->getHotcueNumber();
+    // Note: the cue is just detached from the hotcue control
+    // It remains in the database for later use
+    // TODO: find a rule, that allows us to delete the cue as well
     hotcueClear(pControl, v);
-    CuePointer pCue(m_pLoadedTrack->addCue());
+
+    CuePointer pCue(m_pLoadedTrack->createAndAddCue());
+    double closestBeate = m_pClosestBeat->get();
     double cuePosition =
-            (m_pQuantizeEnabled->get() > 0.0 && m_pClosestBeat->get() != -1) ?
-            floor(m_pClosestBeat->get()) : floor(getCurrentSample());
-    if (!even(static_cast<int>(cuePosition)))
-        cuePosition--;
+            (m_pQuantizeEnabled->toBool() && closestBeate != -1.0) ?
```

Comment:
Typo

And I remember that you once explained comparing a double against a const int might be faster than against a const double. In this file you even replaced some const int with const double in comparisons.